performance_regression_gradual_grow_throughput.py

```diff
@@ -128,34 +108,98 @@ def preload_data(self, compaction_strategy=None):
+                    if key == workload.upper()]
+        self.log.debug("P99 from histogram: %s", p99_list)
+
+        return round(sum(p99_list) / len(p99_list), 2) if p99_list else 0
+
+    def drop_keyspace(self):
+        self.log.debug(f'Drop keyspace {"keyspace1"}')
+        with self.db_cluster.cql_connection_patient(self.db_cluster.nodes[0]) as session:
+            session.execute(f'DROP KEYSPACE IF EXISTS {"keyspace1"};')
+
     # pylint: disable=too-many-arguments,too-many-locals
-    def run_gradual_increase_load(self, stress_cmd_templ,
-                                  start_ops, max_ops, throttle_step,
-                                  stress_num, num_loaders, compaction_strategy, test_name):
-        self.warmup_cache(compaction_strategy)
+    def run_gradual_increase_load(self, stress_cmd_templ, stress_num, num_loaders, compaction_strategy, test_name, sub_type):  # noqa: PLR0914
+        if sub_type in ["read", "mixed"]:
+            self.warmup_cache(stress_cmd_templ)
+
+        if not self.exists():
+            self.log.debug("Create test statistics in ES")
+            self.create_test_stats(sub_type=sub_type, doc_id_with_timestamp=True)
         total_summary = []
-        cs_popuplation_distribution = self.get_cs_distribution()
-        base_stress_cmd = stress_cmd_templ[0].replace("$DIST_PARAMS", cs_popuplation_distribution)
-
-        for current_ops in range(start_ops, max_ops + throttle_step, throttle_step):
-            self.log.info("Run cs command with rate: %s Kops", current_ops)
-            current_throttle = current_ops // (num_loaders * stress_num)
-            stress_cmd = base_stress_cmd.replace("$threads", f"{self.NUM_THREADS}").replace(
-                "$throttle", f"{current_throttle}")
-
-            stress_queue = self.run_stress_thread(stress_cmd=stress_cmd, stress_num=stress_num,
-                                                  compaction_strategy=compaction_strategy, stats_aggregate_cmds=False)
-            results = self.get_stress_results(queue=stress_queue, store_results=False)
-            self.log.debug("All c-s results: %s", results)
-            summary_result = self._calculate_average_latency(results)
-            summary_result["ops"] = current_ops
-            self.log.debug("C-S results for ops: %s. \n Results: \n %s", current_ops, summary_result)
+        num_threads = self.params["perf_threads"][sub_type]
+        throttle_steps = self.params["perf_gradual_throttle_steps"]
+        if sub_type not in throttle_steps:
+            TestFrameworkEvent(source=self.__class__.__name__,
+                               message=f"Throttle steps for '{sub_type}' test is not defined in 'perf_gradual_throttle_steps' parameter",
+                               severity=Severity.CRITICAL).publish()
+        workload = getattr(PerformanceTestWorkload, sub_type.upper())
+
+        for throttle_step in throttle_steps[sub_type]:
+            if sub_type == "write":
+                self.drop_keyspace()
```

Comment:
The "drop" should be in the end of the step, not in the begining, just before the wait for compactions, so we won't wait for compactions.